# Testing the newly made Decision Tree algorithm
I will be using a built in datasets from the sklearn library to set the accuracy of algorithim. There will be 2 datasets I will test for accuracy. 

Test 1: 

In [35]:
from sklearn import datasets
from sklearn.model_selection import train_test_split
import numpy as np
from DT import DecisionTree


data_1 = datasets.load_wine()
#print feauture names
print(data_1.feature_names)
#print target names
print(data_1.target_names)

['alcohol', 'malic_acid', 'ash', 'alcalinity_of_ash', 'magnesium', 'total_phenols', 'flavanoids', 'nonflavanoid_phenols', 'proanthocyanins', 'color_intensity', 'hue', 'od280/od315_of_diluted_wines', 'proline']
['class_0' 'class_1' 'class_2']


In [36]:
X, y = data_1.data, data_1.target

X_train, X_test, y_train, y_test = train_test_split(
    X, y, test_size=0.2, random_state=1234
)

clf = DecisionTree(max_depth=10)
clf.fit(X_train, y_train)
predictions = clf.predict(X_test)

def accuracy(y_test, y_pred):
    return np.sum(y_test == y_pred) / len(y_test)

acc_1 = accuracy(y_test, predictions)
print(acc_1)


0.9444444444444444


Test 2: 

In [28]:
data_2 = datasets.load_iris()
#print feauture names
print(data_2.feature_names)
#print target names
print(data_2.target_names)

X, y = data_2.data, data_2.target

X_train, X_test, y_train, y_test = train_test_split(
    X, y, test_size=0.2, random_state=1234
)

clf = DecisionTree(max_depth=10)
clf.fit(X_train, y_train)
predictions = clf.predict(X_test)

def accuracy(y_test, y_pred):
    return np.sum(y_test == y_pred) / len(y_test)

acc_2 = accuracy(y_test, predictions)
print(acc_2)

['sepal length (cm)', 'sepal width (cm)', 'petal length (cm)', 'petal width (cm)']
['setosa' 'versicolor' 'virginica']
1.0


### Here are the following results for the accuracy of the two datasets: 
#### Wine dataset: 0.9444
#### Iris dataset: 1.0

The Iris dataset has a higher accuracy than the Wine dataset. This is because the Iris dataset has a smaller number of classes compared to the Wine dataset. The Iris dataset has 3 classes while the Wine dataset has 13 classes. The Decision Tree algorithm is able to classify the Iris dataset more accurately because it has fewer classes to distinguish between. The Wine dataset has more classes, which makes it more difficult for the Decision Tree algorithm to classify accurately.